src/DiffOp/Diff/Diff.php

```diff
@@ -12,6 +12,9 @@
  * Base class for diffs. Diffs are collections of DiffOp objects,
  * and are themselves DiffOp objects as well.
  *
+ * @SuppressWarnings(PHPMD.TooManyMethods)
```

Comment:
I was under the impression this was really about public methods and the docs just forget to mention that. I tested now and this is wrong. It's about all methods. In which case I agree with you that this is not a good rule with value 10. It discourages splitting up methods.